src/components/user/unavailability/unavailability.edgedb.repository.ts

```diff
@@ -15,44 +15,44 @@ export class UnavailabilityEdgeDBRepository
     hydrate: (unavailability) => ({
       ...unavailability['*'],
       start: e.assert_exists(e.range_get_lower(unavailability.dates)),
-      end: e.date_range_get_upper(unavailability.dates),
+      end: e.assert_exists(e.range_get_upper(unavailability.dates)),
     }),
   }).customize((cls) => {
     return class extends cls {
       async create(input: CreateUnavailability) {
         const user = e.cast(e.User, e.uuid(input.userId));
-        const unavailability = e.insert(e.User.Unavailability, {
+        const inserted = e.insert(e.User.Unavailability, {
           description: input.description,
-          dates: new Range(input.start, input.end, true, true),
+          dates: new Range(input.start, input.end),
         });
         const updatedUser = e.update(user, () => ({
           set: {
-            unavailabilities: { '+=': unavailability },
+            unavailabilities: { '+=': inserted },
           },
         }));
-        const created = e.assert_exists(
-          e.select(updatedUser.unavailabilities, (u) => ({
-            filter: e.op(u.id, '=', unavailability.id),
-          })),
-        );
-        return await this.db.run(created, this.hydrate);
+        const query = e.select(inserted, (u) => ({
```

Comment:
@CarsonF - So this select statement actually executes the update statement inside of it?

Reply:
I didn't realize you could "daisy chain" statements/queries either; cool.

Reply:
Correct.
```edgeql
with inserted := (insert ...)
select inserted {
  *,
  updatedUser := (update User ...)
}
```
No different than
```edgeql
with updated := (update User ...)
select updated {*}
```

Reply:
🤯